src/Expression/VariableMap.purs

```diff
@@ -0,0 +1,21 @@
+module Expression.VariableMap where
+
+import Prelude
+import Data.Array (find)
+import Data.Maybe (Maybe(..))
+import Data.Tuple (Tuple(..))
+import Expression.Syntax (VariableName)
+
+type VariableMap a
+  = Array (Tuple VariableName a)
```

Comment:
I would normally use Data.Map.  Do you think Data.Map would be an overkill here because the maps will be very small?

Reply:
I did think so. When I initially added it I felt like it would be simpler to implement our own little map.